R/correspondenceanalysis.R

```diff
@@ -238,32 +234,36 @@ CorrespondenceAnalysis = function(x,
         if (output == "Bubble Chart")
         {
             table.maindim <- ifelse(transpose, "columns", "rows")
-            if(is.null(bubble.size))
+            if (is.null(bubble.size))
                 stop("Bubble Charts require bubble sizes.")
-            bubble.size <- matchTableNames(bubble.size, rownames(x), table.maindim)
-            #bubble.names <- rownames(bubble.size)
-         #   if (is.null(bubble.names) && !is.null(names(bubble.size)))
-         #       bubble.names <- names(bubble.size)
-         #   if (is.null(bubble.names))
-         #       stop("The bubble sizes need to be named.")
-         #   bubble.size <- as.numeric(unlist(bubble.size))
-         #   if (length(table.names <- rownames(x)) != length(bubble.names))
-         #       stop("The number of bubble sizes does not match the number of ", table.maindim, " in the table.")
-         #   if (length(unique(bubble.names)) !=  length(bubble.names))
-         #       stop("There are duplicate bubble size names.")
-         #   if (!all(sort(bubble.names) == sort(table.names)))
-         #       if (!all(sort(tolower(bubble.names)) == sort(tolower(table.names))))
-         #       {
-         #           stop("The bubble sizes must contain the same names as in the ",
-         #                table.maindim, " of the input data table: ",
-         #                paste0(paste0(table.names, ":", bubble.names), collapse = ", "), ".")
-         #       }
-            # Sorting bubble sizes to match the row names of the table.
-            #order = match(rownames(x), bubble.names)
-            #if (sum(order, na.rm = TRUE) != sum(1:length(order)))
-            #    stop("The bubble sizes must contain the same names as in the rows of the input data table: ",
-            #         paste0(paste0(table.names, ":", bubble.names), collapse = ", "), ".")
-            #bubble.size = bubble.size[order]
+            bubble.size <- as.matrix(bubble.size)
+            if (is.null(rownames(bubble.size)))
+                stop("The table of bubble sizes need to be named to match the row labels of the input data.")
+            if (NCOL(bubble.size) > 1)
+            {
+                warning("The table of bubble sizes contains more than one column. Only the first column of bubble sizes was used.")
+                bubble.size <- bubble.size[,1,drop = FALSE] 
+            }
+            bubble.size <- TidyTabularData(bubble.size,
+                row.names.to.remove = row.names.to.remove)
+            b.orig.names <- rownames(bubble.size)
+
+            # Modify messages from MatchTable to inform users
+            # that the problem is in bubble.size
+            bubble.size <- InterceptExceptions(MatchTable(bubble.size, ref.names = rownames(x)),
+                error.handler = function(e){
+                    stop("To use a bubble chart, the table of bubble sizes needs to include all the row labels used in the analysis. ", e$message)
```

Comment:
Should you check the error message to see if it actually is due to missing values? If it isn't, just rethrow the error without appending anything? Same with the warning below.

Reply:
Fixed in 52ac5ae